docresources/SyncBot/syncbot-readme.md

```diff
@@ -2,10 +2,120 @@
+       1) Conversion to payload and emission happen as normal
+    1) Logs the success to the console
+    1) Reports any errors to the originating thread
+
+### Specific details
+
+#### Front receives an event
+
+The Front service is configured to send events from the configured inboxes
+via webhook.  These events are then added to the queue for their thread.
+
+#### Front transforms the event into a message
+
+The Front adapter gathers the full event, calculates whether it's the first in
+a thread and which inbox it happened in, bundling this into a generic format.
+
+#### Front transforms the message into a payload
+
+The Front adapter builds a new conversation, message or comment payload,
+depending on what it is given.
+
+#### Front emits the payload
+
+The Front adapter passes to the Front API the payload and passes back some
+basic details.
+
+#### Discourse receives an event
+
+The Discourse service is configured to send events from the configured
+categories via webhook.  These events are then added to the queue for their
+topic.
+
+#### Discourse transforms the event into a message
+
+The Discourse adapter gathers more complete topic and post information,
+bundling this into a generic format.
+
+#### Discourse transforms the message into a payload
+
+The Discourse adapter builds a new topic or post payload, depending on what it
+is given.
+
+#### Discourse emits the payload
+
+The Discourse adapter passes to the Discourse API the payload and passes back
+some basic details.
+
+#### Flowdock receives an event
+
+A Flowdock session is bought online when SyncBot starts, connects to the
+configured rooms and puts any events on the queue for their thread.
+
+#### Flowdock transforms the event into a message
+
+The Flowdock adapter gathers details of the username, and alongside the decent
+details Flowdock provides to the connection bundles it into a generic format
+
+#### Flowdock transforms the message into a payload
+
+The Flowdock adapter creates a new message payload.  This is actually fairly
+simple because one object structure supports all of our purposes.
+
+#### Flowdock emits the payload
+
+The Flowdock adapter passes to the Front API the payload and passes back some
+basic details
+
+#### Listener is quizzed for connected thread
+
+The listener looks back through the message history for a message that matches
+"Connects to {service} thread blah" and returns the blah
+
+#### Username is copied from the existing message details
+
+At this stage usernames across services must match.  This is so that SyncBot
+knows which user's 1-1 history to search for details.  In future this could
+become configurable, in a similar way to flows.
+
+#### Flowdock is quizzed for token
+
+Flowdock will search the 1-1 history of a user for a phrase that matches "My
+{service} token is blah", returning blah
+
+#### Error reporting
+
+Errors, where possible, are reported in the originating thread as whispers.
+
+## Configuration
+
```

Comment:
In the situation where we don't need obfuscation, I don't think files is a bad idea, and in this case, there could actually be a `.procbot.yml` config for SyncBot that lives in `lib/bots/syncbot/.procbot.yml` which SyncBot could load for *everything that is publicly admissible*. The problem still comes down to what needs to be opaque (and do channel names, front inbox names, etc. count as this? Certainly it lets potential attackers know that we use particularly named channels on specific services). For these, we can't easily have a config file, as it'd need to be pulled in from elsewhere. Where? :)